src/agent.js

```diff
@@ -46,48 +51,183 @@ async function generateAllBots(_config) {
   return botConfigs;
 }
 
+/*
+  gatherType "all" is weird. A quick rundown:
+
+  Inbound blocks look like this:
+  HandleBlock
+  - HandleTransaction
+  - HandleTransaction
+  - HandleTransaction
+  - HandleTransaction
+
+  Notable outbound cases:
+
+  for all positive block findings, where we only have block handling bots:
+    HandleBlock -> [Finding, Finding, Finding]
+    - HandleTransaction -> []
+
+  for all positive tx findings, where we only have tx handling bots:
+    HandleBlock -> []
+    - HandleTransaction -> [Finding, Finding, Finding]
+
+  for all positive tx and block findings:
+    HandleBlock -> []
+    - HandleTransaction -> [...blockFindings[Finding], ...txFindings[Finding, Finding]]
+
+  for all positive block findings, some/no tx findings:
+    HandleBlock -> []
+    - HandleTransaction -> []
+
+  for only some positive block findings, all/some/no tx findings:
+    HandleBlock -> []
+    - HandleTransaction -> []
+
+  Bots can have a block handler, a transaction handler, or both
+
+  if there are only bots with block handlers:
+    - run the block handlers, aggregate the results
+    - if all bots return findings, return the findings
+    - else return []
+  if there are only bots with tx handlers:
+    - run the tx handlers, aggregate the results
+    - if all bots return findings, return the findings
+    - else return []
+
+  if you've got mixed tx and block handlers:
+  for all bots with block handlers:
+    - run the handlers, aggregate the results
+    - if all bots return findings, cache that for the tx handlers
+  for all bots with tx handlers:
+    - run the handlers, aggregate the results
+    - if there are no block handlers, and all bots have tx findings, return the findings
+    - if there are block handlers, and all bots have tx findings, return the findings
+*/
+
 async function initialize() {
   const botConfigs = await generateAllBots(config);
+  gatherMode = config.gatherMode;
```

Comment:
Add a quick check of `gatherMode` to ensure that it is defined and either `any` or `all`.